test/com/esotericsoftware/kryo/FieldSerializerTest.java

```diff
@@ -621,7 +621,71 @@ public void testMultipleTimesAnnotatedMapFields () {
 
 		assertFalse("Exception was expected", true);
 	}
-	
+
+	public void testDefaultStrategyForDefaultClass() {
+		TestDefault testDefault = new TestDefault();
+		testDefault.a = "someDefaultValue";
+		kryo.setDefaultSerializer(FieldSerializer.class);
+		kryo.register(TestDefault.class);
+
+		roundTrip(17, 17, testDefault);
+
+		FieldSerializer serializer = (FieldSerializer) kryo.getSerializer(TestDefault.class);
+		assertNotNull(serializer.getField("a"));
+		serializer.removeField("a");
+		assertFieldRemoved(serializer, "a");
+	}
+
+	public void testDefaultStrategyForExtendedClass() {
+		TestExtended testExtended = new TestExtended();
+		((TestDefault) testExtended).a = "someDefaultValue";
+		testExtended.a = "someExtendedValue";
+		kryo.setDefaultSerializer(FieldSerializer.class);
+		kryo.register(TestExtended.class);
+
+		roundTrip(34, 34, testExtended);
+
+		FieldSerializer serializer = (FieldSerializer) kryo.getSerializer(TestExtended.class);
+
+		// the "a" field needs to be removed 2x, once for TestDefault.a and once for TestExtended.a. You
+		// can't remove the second one without removing the first one (in DEFAULT field name strategy)
+		assertNotNull(serializer.getField("a"));
+		serializer.removeField("a");
+		assertNotNull(serializer.getField("a"));
+		serializer.removeField("a");
+		assertFieldRemoved(serializer, "a");
+	}
+
+	public void testExtendedStrategyForExtendedClass() {
+		TestExtended testExtended = new TestExtended();
+		((TestDefault) testExtended).a = "someDefaultValue";
+		testExtended.a = "someExtendedValue";
+		kryo.getFieldSerializerConfig().setCachedFieldNameStrategy(FieldSerializer.CachedFieldNameStrategy.EXTENDED);
+		kryo.setDefaultSerializer(FieldSerializer.class);
+		kryo.register(TestExtended.class);
+
+		roundTrip(34, 34, testExtended);
+
+		FieldSerializer serializer = (FieldSerializer) kryo.getSerializer(TestExtended.class);
+
+		// Simple class name is part of field name in EXTENDED field name strategy.
+		assertNotNull(serializer.getField("TestDefault.a"));
+		serializer.removeField("TestDefault.a");
+		assertFieldRemoved(serializer, "TestDefault.a");
+		assertNotNull(serializer.getField("TestExtended.a"));
+		serializer.removeField("TestExtended.a");
+		assertFieldRemoved(serializer, "TestExtended.a");
+	}
+
+	protected void assertFieldRemoved(FieldSerializer serializer, String fieldName) {
+		try {
+			assertNull(serializer.getField(fieldName));
+			assertTrue(false);

```

Comment:
Could be `Assert.fail("Expected IllegalArgumentException to be thrown for serializer.getField(" + fieldName + ")")`.